service-base/src/main/java/org/eclipse/hono/service/credentials/BaseCredentialsService.java

```diff
@@ -164,6 +179,103 @@ private void processCredentialsMessageGetOperation(final Message<JsonObject> reg
         getCredentials(tenantId, type, authId, result -> reply(regMsg, result));
     }
 
+    private void processCredentialsMessageAddOperation(final Message<JsonObject> regMsg, final String tenantId, final JsonObject payload) {
+        if (!isValidCredentialsObject(payload)) {
```

Comment:
This does not currently check if there was an empty json array for the secrets, does it?
But an add operation with no secrets contained must be avoided IMHO.

Reply:
I thought about this case too, and wasn't really sure about it. I chose to allow empty secret objects, as they seem not to be forbidden according to the specification. Maybe for cases in which the credential type does not require any additional information. E.g. 
```
{
  "device-id": "myDevice",
  "type": "external",
  "auth-id": "external-auth",
  "secrets": [{}]
}
```

But currently no such type exists, so I can add an additional check for this case too.

Reply:
I only meant the array itself, so what we should avoid is `"secrets": []`.
An element with no more keys in it might make sense in special cases (?).
Will change the API documentation at this point to clarify the specification.

Reply:
Yup you're right. I was a bit confused at first, but while implementing i got what you meant. 
Currently it is implement like this: 
- An empty secret is allowed: `"secrets": [{}]` ✔️
- No secrets at all are not allowed `"secrets": []` ❌

Reply:
+1